migrations/2024-11-26-115042_add_combined_tables/up.sql

```diff
@@ -0,0 +1,43 @@
+-- Creates combined tables for the following:
+--
+-- Reports: (comment, post, and private_message)
+-- Inbox: (Comment replies, post replies, comment mentions, post mentions, private messages)
+-- Profile: (Posts and Comments)
+-- Modlog: (lots of types)
+-- Search: (community, post, comment, user, url)
+-- TODO not sure about these two:
+-- Home: (comment, post)
+-- Community: (comment, post)
+CREATE TABLE report_combined (
+    id serial PRIMARY KEY,
+    published timestamptz NOT NULL,
+    post_report_id int REFERENCES post_report ON UPDATE CASCADE ON DELETE CASCADE,
+    comment_report_id int REFERENCES comment_report ON UPDATE CASCADE ON DELETE CASCADE,
+    private_message_report_id int REFERENCES private_message_report ON UPDATE CASCADE ON DELETE CASCADE,
```

Comment:
Foreign key columns should always have indexes

Reply:
Is that necessary here? Shouldn't we only index for read patterns, IE `published desc` ?

Reply:
Indexes on foreign key columns speed up the operations caused by deleting referenced rows or updating the id values. The PaginationCursorData query also needs those indexes.

Reply:
I'm not sure how to do that for the pagination cursor query, but I'll do that for the other columns now.

Reply:
I just realized it can also be done by adding `UNIQUE` individually to each column, which automatically creates the indexes. I don't know if that's better.

Also there's no different index needed for the PaginationCursorData query. The indexes you added are all it needs.

Reply:
> I just realized it can also be done by adding UNIQUE individually to each column, which automatically creates the indexes. I don't know if that's better.

I probably can't do that here since there are a lot of nulls repeated for each column. But I've added the indexes manually anyway.

Reply:
Unique constraint allows multiple null values by default https://www.postgresql.org/docs/current/ddl-constraints.html#DDL-CONSTRAINTS-UNIQUE-CONSTRAINTS

Reply:
K I'll try that instead.